### PR TITLE
Fixes related to supply post, unique buildings and visual bugs

### PR DESCRIPTION
**Pull request by HyperJeanJean**
This is bunch of fixes for some bugs I found lately. Commits descriptions should be explicit enough to understand both bugs and fixes.

I also changed the way the supply post id is refreshed so that it is more consistent with other unique buildings, tell me if you don't want that change.

As always, let me know if you want me to change anything.

I did not test these fixes too much, there might be some regressions I did not notice. I plan to test them a bit more in the near future.

### CODE REVIEW

File: src/building/construction.c [hunk 1]
```diff
@@ -572,7 +574,8 @@ void building_construction_update(int x, int y, int grid_offset)
         building_rotation_get_offset_with_rotation(10, building_rotation_get_rotation(), &x_offset_2, &y_offset_2);
         if (map_building_tiles_are_clear(x, y, 5, TERRAIN_ALL) &&
             map_building_tiles_are_clear(x + x_offset_1, y + y_offset_1, 5, TERRAIN_ALL) &&
-            map_building_tiles_are_clear(x + x_offset_2, y + y_offset_2, 5, TERRAIN_ALL)) {
+            map_building_tiles_are_clear(x + x_offset_2, y + y_offset_2, 5, TERRAIN_ALL) &&
+            !city_buildings_has_hippodrome()) {
```
Comment (crudelios) [inline src/building/construction.c:578]: Can you add this for Julius as a PR too, please? I can do it too but you should get the credit.

**Review by crudelios — COMMENTED** [review]
Just had a look at this. I have some comments/questions for you to check.

File: src/building/construction.c [hunk 2]
```diff
@@ -559,9 +560,10 @@ void building_construction_update(int x, int y, int grid_offset)
     } else if (type == BUILDING_WAREHOUSE) {
         mark_construction(x, y, 3, TERRAIN_ALL, 0);
     } else if (building_is_fort(type)) {
-        if (formation_get_num_legions_cached() < 6) {
+        if (formation_get_num_legions_cached() < formation_get_max_legions()) {
             if (map_building_tiles_are_clear(x, y, 3, TERRAIN_ALL) &&
-                map_building_tiles_are_clear(x + FORT_X_OFFSET[building_rotation_get_rotation()][city_view_orientation()/2], y + FORT_Y_OFFSET[building_rotation_get_rotation()][city_view_orientation()/2], 4, TERRAIN_ALL)) {
+                map_building_tiles_are_clear(x + FORT_X_OFFSET[building_rotation_get_rotation()][city_view_orientation()/2], y + FORT_Y_OFFSET[building_rotation_get_rotation()][city_view_orientation()/2], 4, TERRAIN_ALL) &&
+                city_buildings_has_mess_hall()) {
```
[the remark this answers is not in the document]
Comment (HyperJeanJean) [reply src/building/construction.c:566]: AFAICT, there is already a check for that in ```building_construction_place_building``` (which I also fixed in https://github.com/Keriew/augustus/pull/173/commits/d346ed13b8e257f2469fe8676f819a72ea856781, it did not always work).

File: src/building/construction.c (hunk 1 above)
Comment (HyperJeanJean) [reply src/building/construction.c:578]: Will do. Should I keep this change in this PR as well?

Comment (crudelios) [reply src/building/construction.c:578]: Yes, do keep it here please, it's one less thing to merge later on.

File: src/building/construction.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (crudelios) [reply src/building/construction.c:566]: Humm you're right, then forget about this comment!

**Comment by HyperJeanJean** [comment]
I have tested a bit more the fixes included in this PR and I didn't find any regression. There is still that codefactor complain though, I don't know if I should do something, the function it complains about was already quite long before my additions.

**Comment by crudelios** [comment]
Don't worry about codefactor. I'll have a look at this again and merge if I find no issues.

**Pull request merged**